### DISCUSSION TITLE
WIS WebRTC SDP IP

### DISCUSSION THREAD

**Discussion by FunkyMark8** [toverainc/willow-inference-server#152] [Q&A]
Hi, I've got WIS up&running on PC with RTX 3060 12GB on Ubuntu 22 and I'm starting testing the WebRTC web interface (:19000/rtc). I have quite a lot of experience with VoIP but I'm not familiar with WebRTC. Both PCs with WIS and client computer are on local network (WIS 10.0.130.115, client 10.0.130.110) but in SDP I see the public IP address of internet connection, probably due to this it ends with iceConnectionLog failed
Disconnected from ASR Service

Could you please give an advice?

Many thanks in advance!

The complete SDP:
  "sdp": "v=0\r\no=mozilla...THIS_IS_SDPARTA-99.0 2032908331779630016 0 IN IP4 0.0.0.0\r\ns=-\r\nt=0 0\r\na=sendrecv\r\na=fingerprint:sha-256 0B:AA:2D:D2:60:B8:7E:C4:0E:7A:28:5D:87:D9:9D:65:64:00:84:BC:60:91:41:91:17:97:12:8A:21:90:2B:6B\r\na=group:BUNDLE 0 1\r\na=ice-options:trickle\r\na=msid-semantic:WMS *\r\nm=audio 23339 UDP/TLS/RTP/SAVPF 109 9 0 8 101\r\nc=IN IP4 109.81.xxx.xxx\r\na=candidate:1 1 UDP 1686052863 109.81.xxx.xxx 23339 typ srflx raddr 109.81.xxx.xxx rport 23339\r\na=candidate:1 2 UDP 1686052862 109.81.xxx.xxx 23466 typ srflx raddr 109.81.xxx.xxx rport 23466\r\na=sendrecv\r\na=end-of-candidates\r\na=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level\r\na=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level\r\na=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid\r\na=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1\r\na=fmtp:101 0-15\r\na=ice-pwd:38115d6433b869d24edc525dc76894ae\r\na=ice-ufrag:53b269bc\r\na=mid:0\r\na=msid:- {b871a472-7166-4991-a31d-74e744004a27}\r\na=rtcp:23466 IN IP4 109.81.xxx.xxx\r\na=rtcp-mux\r\na=rtpmap:109 opus/48000/2\r\na=rtpmap:9 G722/8000/1\r\na=rtpmap:0 PCMU/8000\r\na=rtpmap:8 PCMA/8000\r\na=rtpmap:101 telephone-event/8000\r\na=setup:actpass\r\na=ssrc:1566551442 cname:{25cf97ef-7106-4393-8310-a28ff99ae4f4}\r\nm=application 23379 UDP/DTLS/SCTP webrtc-datachannel\r\nc=IN IP4 109.81.xxx.xxx\r\na=candidate:1 1 UDP 1686052863 109.81.xxx.xxx 23379 typ srflx raddr 109.81.xxx.xxx rport 23379\r\na=sendrecv\r\na=end-of-candidates\r\na=ice-pwd:38115d6433b869d24edc525dc76894ae\r\na=ice-ufrag:53b269bc\r\na=mid:1\r\na=setup:actpass\r\na=sctp-port:5000\r\na=max-message-size:1073741823\r\n"

**Comment by kristiankielhofner** [comment]
There are a couple of ways to address this:

1) Disable STUN in client and server to prevent discovery of externally reachable IP addresses. This isn't recommended because it requires various code changes on server and client and will prevent any use of WIS WebRTC outside the local network. We've investigated supporting this with relatively simple configuration changes but haven't settled on an approach to do so.

2) Forward configured WebRTC UDP port ranges on your router to the WIS host. This configuration will require compatible support on your router. Default WIS WebRTC dynamic port range is UDP 10000-10050. You've likely seen this approach with dynamically allocated RTP ports in your VoIP experience. It's not as much of a security concern as typically believed because the sockets are allocated dynamically and connections are only allowed from IP:PORT pairs based on ICE/SDP negotiation.
